src/rngs/thread.rs

```diff
@@ -42,17 +42,30 @@ const THREAD_RNG_RESEED_THRESHOLD: u64 = 1024 * 64;
 ///
 /// This type is a reference to a lazily-initialized thread-local generator.
 /// An instance can be obtained via [`thread_rng`] or via `ThreadRng::default()`.
-/// This handle is safe to use everywhere (including thread-local destructors),
-/// though it is recommended not to use inside a fork handler.
 /// The handle cannot be passed between threads (is not `Send` or `Sync`).
 ///
 /// `ThreadRng` uses the same CSPRNG as [`StdRng`], ChaCha12. As with
 /// [`StdRng`], the algorithm may be changed, subject to reasonable expectations
 /// of security and performance.
 ///
 /// `ThreadRng` is automatically seeded from [`OsRng`] with periodic reseeding
-/// (every 64 kiB, as well as "soon" after a fork on Unix — see [`ReseedingRng`]
-/// documentation for details).
+/// (every 64 kiB — see [`ReseedingRng`] documentation for details).
+///
+/// `ThreadRng` is not automatically reseeded on fork. It is recommended to
+/// explicitly call [`ThreadRng::reseed`] immediately after a fork, for example:
```

Comment:
It also could be worth to mention `pthread_atfork` as  a potential approach for doing this.

Reply:
Technically, it might be okay: `ThreadRng` methods should not be called at the time `fork()` happens. But I don't see the need to advertise this beyond what is already said.